### DISCUSSION TITLE
Setting up vortex with self-hosted give InvalidOperation error

### DISCUSSION THREAD

**Discussion by tsukasaroot** [stoatchat/discussions#211] [Q&A] [answered]
Hello, I could setup revoltchat through docker with https and everything runs smoothly on this side. I then decided to use vortex, so builded vortex image, then configured .env and my vhost to make the connection. 

I can access the url https://comm.digitalsavior.fr/vortex and https://comm.digitalsavior.fr/vortex-ws seems functionnal too as it gives same output as https://comm.digitalsavior.fr/ws when accessing it.

Though when i try to join a vocal chat, I get 400 bad request "InvalidOperation" immediately after clicking call button in a channel.

Here is my vhost :

<VirtualHost *:443>

	ServerName comm.digitalsavior.fr
	SSLEngine on
	SSLCertificateFile "C:\gogs\custom\https\unified.cert"
	SSLCertificateKeyFile "C:\gogs\custom\https\decryped-private.key"
	SSLCertificateChainFile "C:\gogs\custom\ca.crt"
	SSLProxyEngine on
	SSLProxyVerify none
	SSLProxyCheckPeerCN off
	SSLProxyCheckPeerName off
	SSLProxyCheckPeerExpire off
	
	<Location "/">
		ProxyPreserveHost On
		ProxyPass http://localhost:5000/
	</Location>
	
	<Location "/api">
	    ProxyPreserveHost On
		ProxyPass http://localhost:8000/
	</Location>
	
	<Location "/autumn">
	    ProxyPreserveHost On
		ProxyPass http://localhost:3000/
	</Location>

	<Location "/january">
	    ProxyPreserveHost On
		ProxyPass http://localhost:7000/
	</Location>

	<Location "/ws">
		ProxyPreserveHost On
		ProxyPass ws://localhost:9000/
		ProxyPassReverse ws://localhost:9000/
		RewriteEngine on
		#RewriteCond %{HTTP:Upgrade} websocket [NC]
		#RewriteCond %{HTTP:Connection} upgrade [NC]
		#RewriteRule ^/?(.*) "ws://localhost:9000/$1" [P,L]
	</Location>
	
	<Location "/vortex">
	    ProxyPreserveHost On
		ProxyPass http://localhost:8080/
		RewriteEngine on
		RewriteCond %{HTTP:Upgrade} websocket [NC]
		RewriteCond %{HTTP:Connection} upgrade [NC]
		RewriteRule ^/?(.*) "ws://localhost:8080/$1" [P,L]
	</Location>

	<Location "/vortex-ws">
		ProxyPreserveHost On
		ProxyPass ws://localhost:8080/
		RewriteEngine on
		RewriteCond %{HTTP:Upgrade} websocket [NC]
		RewriteCond %{HTTP:Connection} upgrade [NC]
		RewriteRule ^/?(.*) "ws://localhost:8080/$1" [P,L]
	</Location>

</VirtualHost>

Here my .env 
[env.txt](https://github.com/revoltchat/revolt/files/7455404/env.txt)

and my docker file part about vortex :
  # REVOLT voice server (Vortex)
  vortex:
    image: vortex
    env_file: .env
    ports:
      - "8080:8080"
      - "$RTC_MIN_PORT-$RTC_MAX_PORT:$RTC_MIN_PORT-$RTC_MAX_PORT/tcp"
      - "$RTC_MIN_PORT-$RTC_MAX_PORT:$RTC_MIN_PORT-$RTC_MAX_PORT/udp"

Is there anything that seems off and would cause this issue ? (ofc ports are opened in router and firewall of server both tcp/udp)
I also tried with the revoltchat/vortex:0.3.0-alpha.1 but same error :/

thanks alot!

**Comment by fatalerrorcoded** [comment, marked as answer]
Your webserver configuration appears to be misconfigured. I don't know how it's routing as I don't do Apache, but the room management endpoint (`/room/roomid` as should be seen from vortex) returns a Not Found, suggesting that when delta tries to create a new room by calling `/vortex/room/roomid`, it fails because Apache probably isn't correctly stripping off the `vortex/` part, thus erroring out with `InvalidOperation`

**Reply by tsukasaroot** [reply]
when you write (as should be seen from vortex), you mean vortex logs right? In the docker, i do not have any output on this one, beside the listening part
[2021-11-01T19:25:25Z INFO  vortex] Starting Revolt Vortex voice server
[2021-11-01T19:25:25Z INFO  warp::server] Server::run; addr=0.0.0.0:8080
[2021-11-01T19:25:25Z INFO  warp::server] listening on http://0.0.0.0:8080 
When I do call action, API returns a success message though, so thought it was fine

[2021-11-01T19:33:05Z INFO  rocket::server] POST //channels/01FKDNSTK3WJBQR5Y7VMQ0AX2T/join_call:
[2021-11-01T19:33:05Z INFO  rocket::server] Matched: (req) POST /channels/<target>/join_call
[2021-11-01T19:33:06Z INFO  rocket::server] Outcome: Success
[2021-11-01T19:33:06Z INFO  rocket::server] Response succeeded.

So maybe you can confirm that my vortex apache config is completely wrong?

**Reply by tsukasaroot** [reply]
I think you are correct, tho i don't have the skill to find out how to fix this url with apache, so I moved the VOSO_PUBLIC_URL on a subdomain, made a vhost for it redirecting to http://localhost:8080/ correctly and now it seems functional.
Thanks for your time!

**Reply by fatalerrorcoded** [reply]
What I think is the issue is that Apache's configured to add an extra slash in the ProxyPass entries (they probably shouldn't end with a slash), seeing that delta is getting requests such as `//channels/01FKDNSTK3WJBQR5Y7VMQ0AX2T/join_call`, and while rocket handles it just fine, warp (the framework vortex uses) probably doesn't.

**Reply by tsukasaroot** [reply]
Dang, thanks for the log update I could understand where was the issue with the uri!!
So for those using apache and having this issue
by using the original setup on the post with the all in one domain, we only need to add a '/' at the end of the location match like so : 

<Location "/vortex/">

Thanks again for your time @fatalerrorcoded !! :)

EDIT:
I also had this issue with autumn, so gave it a try and adding trailing slash at the end of location match does work too!
